src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -534,17 +533,18 @@ protected virtual IEnumerable<string> GetDependsOnRequestIds(IEnumerable<string>
         /// <param name="contentId">The contentId of this request message.</param>
         /// <param name="groupId">The group id that this request belongs to. Can be null.</param>
         /// <param name="dependsOnIds">The prerequisite request ids of this request.</param>
-        /// <param name="batchFormat">Format of the batch.</param>
+        /// <param name="dependsOnIdsValidationRequired">Whether the <code>dependsOnIds</code> value needs to be validated.</param>
         /// <returns>An <see cref="ODataBatchOperationRequestMessage"/> to write the request content to.</returns>
         protected ODataBatchOperationRequestMessage BuildOperationRequestMessage(Stream outputStream, string method, Uri uri,
-            string contentId, string groupId, IEnumerable<string> dependsOnIds, ODataFormat batchFormat)
+            string contentId, string groupId, IEnumerable<string> dependsOnIds, bool dependsOnIdsValidationRequired)
         {
             IEnumerable<string> flattenDependsOnIds = dependsOnIds == null
                 ? null
                 : GetDependsOnRequestIds(dependsOnIds);
 
-            if (flattenDependsOnIds != null)
+            if (flattenDependsOnIds != null && dependsOnIdsValidationRequired)
             {
+                // Validate dependsOn ids in Json batch.

```

Comment:
Base class should not refer to derived classes, even in comments.  We validate dependsOnIds if dependsOnIdsValidationRequired is true.  Why it's true is outside of the scope of this class.

Reply:
Please remove references to JSON and Multipart from comments in the base class.  Comment should read something like "Validate dependsOnIds if specified".

---
In reply to: [158349752](https://github.com/OData/odata.net/pull/1020#discussion_r158349752) [](ancestors = 158349752)

Reply:
done for base classes reader and writer. 

---
In reply to: [160554679](https://github.com/OData/odata.net/pull/1020#discussion_r160554679) [](ancestors = 160554679,158349752)